lib/api/initiateMultipartUpload.js

```diff
@@ -258,47 +260,50 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
 
     metadataValidateBucketAndObj(metadataValParams, log,
         (err, destinationBucket) => {
-            const corsHeaders = collectCorsHeaders(request.headers.origin,
-                request.method, destinationBucket);
-            if (err) {
-                log.debug('error processing request', {
-                    error: err,
-                    method: 'metadataValidateBucketAndObj',
-                });
-                return callback(err, null, corsHeaders);
-            }
-            if (destinationBucket.hasDeletedFlag() &&
-                accountCanonicalID !== destinationBucket.getOwner()) {
-                log.trace('deleted flag on bucket and request ' +
-                    'from non-owner account');
-                return callback(errors.NoSuchBucket);
-            }
-            if (destinationBucket.hasTransientFlag() ||
-                destinationBucket.hasDeletedFlag()) {
-                log.trace('transient or deleted flag so cleaning ' +
-                    'up bucket');
-                return cleanUpBucket(destinationBucket,
-                        accountCanonicalID, log, err => {
-                            if (err) {
-                                log.debug('error cleaning up bucket ' +
-                                    'with flag',
-                                    { error: err,
-                                        transientFlag:
-                                    destinationBucket.hasTransientFlag(),
-                                        deletedFlag:
-                                    destinationBucket.hasDeletedFlag(),
-                                    });
-                                // To avoid confusing user with error
-                                // from cleaning up
-                                // bucket return InternalError
-                                return callback(errors.InternalError,
-                                    null, corsHeaders);
-                            }
-                            return _storetheMPObject(destinationBucket,
-                                corsHeaders);
+            getObjectSSEConfiguration(request.headers, destinationBucket, log,
+                (err, objectSSEConfig) => {
+                    const corsHeaders = collectCorsHeaders(request.headers.origin,
+                        request.method, destinationBucket);
+                    if (err) {
```

Comment:
This error handling block should be executed before calling `getObjectSSEConfiguration`, and probably there should be an extra block here to check errors from this function